larq_compute_engine/core/packbits.h

```diff
@@ -301,16 +321,28 @@ inline void pack_optimized(const float* in, std::uint64_t* out) {
 
 // Helper function
 template <BitpackOrder bitpack_order, class TIn, class TOut>
-inline void pack_bitfield(const TIn* in, TOut* out) {
-  if (bitpack_order == BitpackOrder::Canonical)
-    pack_canonical(in, out);
-  else
-    pack_optimized(in, out);
+inline void pack_bitfield(const TIn* in, TOut* out,
+                          const std::int32_t zero_point) {
```

Comment:
by defining default value for `zero_point` the user does not need to pass this argument if its not using this function for non-quantized inputs.

Reply:
For this function, I deliberately left it like this, because its an internal function (we only use it inside `packbits.h`) so this way we can't accidentally forget to pass the `zero_point` argument. For a function like `packbits_matrix`, which is used everywhere, I did set the `zero_point` default value, so that it would be cleaner and easier to use.
I don't feel too strongly about this, so I'm also fine with setting `zero_point=0` by default here.

Reply:
it seems to me this function accepts general types (both quantinzed and non-quantized inputs), therefore I think for the general functions it makes sense to set the default value but specialized functions I agree with you.

Reply:
oh wait. Nevermind. I think here your approach is better.